modules/core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/StreamWorkManager.java

```diff
@@ -296,45 +295,41 @@ public void init() {
             if (started) {
                 return;
             }
-            getDescriptors(QUEUES_EP).forEach(d -> categoryToQueueId.put(d.getId(), d.getId()));
+            getWorkQueueIds().forEach(id -> categoryToQueueId.put(id, id));
             index();
             initTopology();
             logManager = getLogManager();
             streamManager = getStreamManager();
             streamManager.register("StreamWorkManagerDisable", topologyDisabled, settings);
             streamProcessor = streamManager.registerAndCreateProcessor("StreamWorkManager", topology, settings);
             started = true;
-            new ComponentListener().install();
-            log.info("Initialized");
-        }
-    }
 
-    class ComponentListener implements ComponentManager.Listener {
-        @Override
-        public void beforeStop(ComponentManager mgr, boolean isStandby) {
-            if (!shutdown(10, TimeUnit.SECONDS)) {
-                log.error("Some processors are still active");
-            }
-        }
+            Framework.getRuntime().getComponentManager().addListener(new ComponentManager.Listener() {
```

Comment:
Could you add a comment explaining why it's not possible to make the component itself implement `ComponentManager.Listener`?

Reply:
I gave up on making that change in a preceding PR (see https://github.com/nuxeo/nuxeo/pull/4544#issuecomment-749534696). The fact that StreamWorkManager is supposed to replace the WorkManagerImpl component is highly unusual for the runtime, and my understanding is that works should be removed anyway so I did not dig further. I can still add this as a comment.